paper/paper.md

```diff
@@ -0,0 +1,327 @@
+from pymor.core.rule import Rule
+
+# Note that type hints are encouraged (but not enforced) by PyMOR
+def add_pp_components(data: xr.Dataset, rule: Rule) -> xr.DataArray:
+    """Adds together Net Primary Production components"""
+    # In this example, we utilize two different input variable in the native
+    # model outputs with cryptic names to construct a well-known standard name
+    # for the Net Primary Production (NPP) variable.
+    data["pp"] = data["diags3d01"] + data["diags3d02"]
+    # This return is mandatory for the pipeline to usefully continue
+    # We return our new variable to be further processed:
+    return data["pp"]  # Return type: DataArray.
+```
+
+Then, the user adds the path to the file containing this function to their
+pipeline definition:
+
+```yaml
+pipelines:
+  - name: convert_with_pp
+    steps:
+      - "pymor.core.gather_inputs.load_mfdataset"
+      - "script://./intpp_recom.py:add_pp_components"
+      - "pymor.fesom_1p4.nodes_to_levels"
+      - "script://./intpp_recom.py:vertical_integration"
+      - "script://./intpp_recom.py:set_pp_units"
+      - "pymor.std_lib.convert_units"
+      - "pymor.std_lib.time_average"
+      - "pymor.std_lib.set_global_attributes"
+      - "pymor.std_lib.trigger_compute"
+      - "pymor.std_lib.show_data"
+      - "pymor.std_lib.files.save_dataset"
+```
+
+The special tag `script://` followed by an absolute path to a Python file and
+the function name separated by a colon allows user-defined functionality, as:
+
+```yaml
+pipelines:
+   - name: <some_name>
+     steps:
+        - "script://<path/to/some/script.py>:<function_name>"
+```
+
+**Parallelization and scalability** is achieved by using
+[`dask`](https://dask.org/) and
+[`dask-jobqueue`](https://jobqueue.dask.org/en/latest/) to distribute the
+workload in a High Performance Computing (HPC) environment, typically in such a
+way that the users can run post-processing pipelines on the same machine that
+they produced the simulations with. Simple usage of dask is enabled directly in
+the user configuration file:
+
+```yaml
+pymor:
+  parallel: True
+  dask_cluster: "slurm"
+  dask_cluster_scaling_mode: fixed
+  fixed_jobs: 12
+```
+
+**Simple command line usage** enables launching all conversion
+functionality once the user has defined the configuration file:
+
+```bash
+$ pymor process <path/to/config.yaml>
+```
+
+Once launched, simple monitoring is available to examine both the Dask
+processing, as well as the conversion of individual pipelines, which are
+optionally orchestrated via [`prefect`](https://prefect.io/). When using an
+orchestrator, the user also has options to cache intermediate results, thus
+enabling only partial processing of the conversion if the process needs to be
+restarted for whatever reason.
+
+# Acknowledgements
+
+We acknowledge valuable contributions from Christopher Danek and
+Christian Stepanek for early design testing, as well as the CMIP team for
+fruitful discussions and feedback.
```

Comment:
```suggestion
Christian Stepanek for early design testing, as well as the CMIP team and the WCRP ESMO Infrastructure Panel (WIP) fruitful discussions and feedback.
```
We also need to acknowledge here the DataHub which is the funding body for @siligam's position